app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt

```diff
@@ -421,4 +432,28 @@ class CalculatorImpl(
     private fun Double.format() = formatter.doubleToString(this)
 
     private fun String.removeGroupSeparator() = formatter.removeGroupingSeparator(this)
+
+
+    fun getCalculatorStateJson(): JSONObject {
+        val jsonObj = JSONObject()
+        jsonObj.put(RES, currentResult)
+        jsonObj.put(PREVIOUS_CALCULATION, previousCalculation)
+        jsonObj.put(LAST_KEY, lastKey)
+        jsonObj.put(LAST_OPERATION, lastOperation)
+        jsonObj.put(BASE_VALUE, baseValue)
+        jsonObj.put(SECOND_VALUE, secondValue)
+        jsonObj.put(INPUT_DISPLAYED_FORMULA, inputDisplayedFormula)
+        return jsonObj
+    }
+
+    private fun setFromSaveInstanceState(json: String) {
+        val jsonObject = JSONTokener(json).nextValue() as JSONObject
+        currentResult = jsonObject.getString("res")
```

Comment:
you should use the constants here too, that is the point of it. To avoid issues caused by typos.

Reply:
Sure, it's done.